services/motion/explore/explore.go

```diff
@@ -324,36 +334,66 @@ func (ms *explore) checkForObstacles(
 	plan motionplan.Plan,
 	obstaclePollingFrequencyHz float64,
 ) {
+	ms.logger.Debug("Current Plan")
+	for i, p := range plan {
+		ms.logger.Debugf("plan[%v]: %v", i, p)
+	}
 	// Constantly check for obstacles in path at desired obstacle polling frequency
 	ticker := time.NewTicker(time.Duration(int(1000/obstaclePollingFrequencyHz)) * time.Millisecond)
 	defer ticker.Stop()
 
+	var errCounterCurrentInputs, errCounterGenerateTransientWorldState int
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			currentPose := spatialmath.NewZeroPose()
+
+			currentInputs, err := kb.CurrentInputs(ctx)
+			if err != nil {
+				ms.logger.Debugf("issue occurred getting current inputs from kinematic base: %v", err)
+				if errCounterCurrentInputs > successiveErrorLimit {
+					ms.obstacleResponseChan <- moveResponse{success: false}
+					return
+				}
+				errCounterCurrentInputs += 1
+			} else {
+				errCounterCurrentInputs = 0
+			}
+
+			currentPose := spatialmath.NewPose(
+				r3.Vector{X: currentInputs[0].Value, Y: currentInputs[1].Value, Z: 0},
+				&spatialmath.OrientationVector{OX: 0, OY: 0, OZ: 1, Theta: currentInputs[2].Value},
+			)
 
 			// Look for new transient obstacles and add to worldState
 			worldState, err := ms.generateTransientWorldState(ctx, obstacleDetectors)
 			if err != nil {
 				ms.logger.Debugf("issue occurred generating transient worldState: %v", err)
+				if errCounterGenerateTransientWorldState > successiveErrorLimit {
```

Comment:
For both here and above, what sort of errors are we expecting to see that we want to ignore? Why are there errors and why is it okay to ignore? Can we add this as a comment as well?

Reply:
The only errors that will appear here are:
- [generateTransientWorldState] issue with vision service
- [generateTransientWorldState] duplicate geometries detected (unlikely as this would mean the vision service returned multiple of the same detections
- [CurrentInputs] None when localizer is empty, but possible if current position fails when the localizer is present

All of these are unlikely. If they do occur they could be permanent or transient. By ignoring them we allow transient issues to no affect operation. That being said, I don't feel too strongly about this as I odnt expect these errors to happen often.